R/helper.R

```diff
@@ -1,3 +1,55 @@
+#' @title 
+#' Extract and format parameter values for `PartInv`.
+#' 
+#' @name 
+#' format_cfa_partinv
+#'
+#' @description
+#' \code{format_cfa_partinv} takes in a lavaan CFA fit object and a component
+#'  and returns the necessary inputs for PartInv in a list
+#'  
+#' @param obj lavaan CFA output
+#'  
+#' @return The output will be a list of 5 elements:
+#'    \item{nu}{A list of length `g` containing `1 x n` measurement intercept
+#'     vectors where `g` is the number of groups and `n` is the number of items
+#'     in the scale.}
+#'    \item{alpha}{A list of length `g` containing `1 x d` latent factor mean
+#'     vectors where `g` is the number of groups and `d` is the number of 
+#'     latent dimensions.}
+#'    \item{lambda}{A list of length `g` containing `n x d` factor loading
+#'     matrices where `g` is the number of groups, `d` is the number of 
+#'     latent dimensions, and `n` is the number of items in the scale.}
+#'    \item{psi}{A list of length `g` containing `d x d` latent factor
+#'     variance-covariance matrices where `g` is the number of groups and `d` 
+#'     is the number of latent dimensions.}
+#'    \item{Theta}{A list of length `g` containing `1 x n` vectors or `n x n`
+#'     matrices of unique factor variances and covariances, where `g` is the
+#'     number of groups and `n` is the number of items in the scale.}
+#'     
+#' @export
+format_cfa_partinv <- function(obj, comp) {
```

Comment:
The function is fine. I just think we could also define a more generic function like

```r
unnest_list <- function(ins) {
        nms <- names(ins[[1]])
        num_gr <- length(ins)
        out <- rep(list(vector(mode = "list", length = num_gr)), length(nms))
        names(out) <- nms
        for (i in seq_along(ins)) {
            for (nm in nms) {
                out[[nm]][[i]] <- ins[[i]][[nm]]
            }
        }
        out
    }
```

and then have your function call this function directly:

```r
unnest_list(lavInspect(fit, what = "est"))
```

Reply:
Looks great, thank you. I have added this function to helper.R with documentation.

Reply:
My thought is that you can just call `unnest_list()` inside `format_cfa_partinv()`